### PR TITLE
feat(EMS-814): Company Details - Turnover - Save and go back functionality

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:
* Added save and go back functionality for turnover page
* Added database fields in exporterBusiness table for turnover data
* Added tests
* Renamed save.natureOfYourBusiness to save.exporterBusiness as both "nature of your business" and "turnover" are both saved to the exporterBusiness table.  exporterCompany stays the same

### CODE REVIEW

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/turnover/save-and-back.spec.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    it('should retain the `your business` task status as `in progress`', () => {
+      const expected = TASKS.STATUS.IN_PROGRESS;
+      cy.checkText(task.status(), expected);
+    });
+
+    it(`should retain the ${ESTIMATED_ANNUAL_TURNOVER} input on the page and the other fields should be empty`, () => {
+      task.link().click();
+      submitButton().click();
+      submitButton().click();
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/journeys/insurance/your-business/turnover/save-and-back.spec.js:89]: Can we add a comment to explain why there are 2 submit button clicks here

File: src/ui/server/controllers/insurance/business/map-and-save/index.test.ts [hunk 2]
```diff
@@ -143,4 +145,65 @@ describe('controllers/insurance/business/map-and-save', () => {
...
+
+  describe('turnover', () => {
+    let mockFormBody = {
+      _csrf: '1234',
+      [PERCENTAGE_TURNOVER]: '25',
+      [ESTIMATED_ANNUAL_TURNOVER]: '35000',
+    };
+
+    const mockSaveExporterBusiness = jest.fn(() => Promise.resolve({}));
+    save.exporterBusiness = mockSaveExporterBusiness;
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/map-and-save/index.test.ts:157]: Seems it should be `save.turnover`?

File: src/ui/server/controllers/insurance/business/nature-of-business/map-submitted-data/index.test.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { RequestBody } from '../../../../../../types';
 import { FIELD_IDS } from '../../../../../constants';
 import mapSubmittedData from '.';
-import { mockExporterBusiness } from '../../../../../test-mocks';
+import { mockExporterBusinessNatureOfBusiness } from '../../../../../test-mocks';
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/nature-of-business/map-submitted-data/index.test.ts:4]: `ExporterNatureOfBusiness`?

File: src/ui/server/controllers/insurance/business/turnover/save-and-back/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('post - save and back', () => {
+    describe('when there are no validation errors', () => {
+      it('should redirect to all sections page', async () => {
+        req.body = {
+          ...mockExporterBusinessTurnover,
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/turnover/save-and-back/index.test.ts:37]: No need for the spread operator in this test file - duplicated below 

Could also move to a `validBody` const

File: src/ui/server/test-mocks/mock-exporter-business-nature-of-business.ts [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
-const mockExporterBusiness = {
+const mockExporterBusinessNatureOfBusiness = {
   goodsOrServicesSupplied: 'ABC',
   totalYearsExporting: '2,0',
   totalEmployeesInternational: '1,000',
   totalEmployeesUK: '4,00',
 } as ExporterBusiness;
 
-export default mockExporterBusiness;
+export default mockExporterBusinessNatureOfBusiness;
```
Comment (ttbarnes) [inline src/ui/server/test-mocks/mock-exporter-business-nature-of-business.ts:10]: Can we align all the test mocks here to be inline with the DB structure/what would get returned from the API? We've been doing that for other areas e.g

```js
const mockApplication = {
  // ...
  policyAndExports: { ... },
  exporterCompany: { ... },
  exporterBusiness: { ... },
};
```

I.e there's no "business turnover or "nature of business" in the structure it's all under `exporterBusiness`

Comment (Zainzzkk) [reply src/ui/server/test-mocks/mock-exporter-business-nature-of-business.ts:10]: I am using these mocks for what the body could be, if I put them all in one, the test would not be representative of the update (turnover would not get natureOfBusiness fields)

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/turnover/turnover-page.spec.js [hunk 6]
```diff
@@ -136,4 +139,22 @@ context('Insurance - Your business - Turnover page - As an Exporter I want to en
 
     cy.checkText(saveAndBackButton(), BUTTONS.SAVE_AND_BACK);
   });
+
+  describe('form submission', () => {
+    it(`should redirect to ${BROKER}`, () => {
+      cy.completeAndSubmitTurnover();
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply e2e-tests/cypress/e2e/journeys/insurance/your-business/turnover/turnover-page.spec.js:145]: Marked as resolved but no `Form` here?

File: src/ui/server/controllers/insurance/business/turnover/index.test.ts [hunk 7]
```diff
@@ -137,6 +142,19 @@ describe('controllers/insurance/business/turnover', () => {
         const expected = `${INSURANCE_ROOT}/${mockApplication.referenceNumber}${BROKER_ROOT}`;
         expect(res.redirect).toHaveBeenCalledWith(expected);
       });
+
+      it('should call mapAndSave.companyDetails once with natureOfBusiness and application', async () => {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/turnover/index.test.ts:146]: typo(s) here seems it should be `mapAndSave.turnover`and without `natureOfBusiness`?

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/turnover/turnover-page.spec.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply e2e-tests/cypress/e2e/journeys/insurance/your-business/turnover/turnover-page.spec.js:145]: I did the wrong line whoops :P

File: src/ui/server/test-mocks/mock-exporter-business-nature-of-business.ts (hunk 5 above)
Comment (ttbarnes) [reply src/ui/server/test-mocks/mock-exporter-business-nature-of-business.ts]: Where are these mocks used in this manner? I can't see

Beacuse the idea is this mock application is what we'd get from the API i.e

```js
application {
  policyAndExport { ... },
  exporterCompany: { ... },
  exporterBusiness: { ... }
}
```

The nature of business / turnover fields are inside `exporterBusiness` and doesn't the actual api/DB update only  `exporterBusiness`? Bit confused why it's structured differently in the mocks.

File: src/ui/server/helpers/mappings/map-application-to-form-fields/index.ts [hunk 8]
```diff
@@ -64,6 +64,8 @@ const mapApplicationToFormFields = (application: Application): object => {
         [YEARS_EXPORTING]: transformNumberToString(application.exporterBusiness[YEARS_EXPORTING]),
         [EMPLOYEES_UK]: transformNumberToString(application.exporterBusiness[EMPLOYEES_UK]),
         [EMPLOYEES_INTERNATIONAL]: transformNumberToString(application.exporterBusiness[EMPLOYEES_INTERNATIONAL]),
+        [PERCENTAGE_TURNOVER]: transformNumberToString(application.exporterBusiness[PERCENTAGE_TURNOVER]),
+        [ESTIMATED_ANNUAL_TURNOVER]: transformNumberToString(application.exporterBusiness[ESTIMATED_ANNUAL_TURNOVER]),
```
Comment (ttbarnes) [inline src/ui/server/helpers/mappings/map-application-to-form-fields/index.ts:68]: Update unit test please

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**